From ebefd77cceacdfcb987abdc62be587aa584f12ea Mon Sep 17 00:00:00 2001
From: Jakub Jelen <jjelen@redhat.com>
Date: Tue, 27 Oct 2026 12:35:00 +0100
Subject: [PATCH] Verify source signatures with openpgpverify

Replace the GnuPG-based %gpgverify with %openpgpverify, which uses
Sequoia sqv and takes the same options, as part of the Fedora 45 Change
https://fedoraproject.org/wiki/Changes/Sequoia_openpgpverify

Refresh Public-Key from keyservers/WKD: same certificate(s), with
self-signatures our copy was missing; the commands are in the spec
comment.
Release bumped with a changelog entry.

Assisted-by: Claude Opus 5.5
---
 mingw-pcre.spec | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/mingw-pcre.spec b/mingw-pcre.spec
index 131807f..3d5e0de 100644
--- a/mingw-pcre.spec
+++ b/mingw-pcre.spec
@@ -7,13 +7,17 @@
 Name:		mingw-%{name1}
 Version:	8.45
 %global myversion %{version}%{?rcversion:-%rcversion}
-Release:	11%{?dist}
+Release:	12%{?dist}
 Summary:	MinGW Windows pcre library
 
 License:	BSD-3-Clause
 URL:		http://www.pcre.org/
 Source0:        https://ftp.pcre.org/pub/%{name1}/%{?rcversion:Testing/}%{name1}-%{myversion}.tar.bz2
 Source1:        https://ftp.pcre.org/pub/%{name1}/%{?rcversion:Testing/}%{name1}-%{myversion}.tar.bz2.sig
+# Refreshed from keyservers/WKD (same keys, newer signatures):
+#   sq --home=none network search --output - 45F68D54BBE23FB3039B46E59766E084FB0F43D8 | sq keyring filter --experimental --cert 45F68D54BBE23FB3039B46E59766E084FB0F43D8 --output update.pgp
+#   sq keyring merge Public-Key update.pgp | sq packet dearmor --output Public-Key.new
+#   mv Public-Key.new Public-Key
 Source2:        https://ftp.pcre.org/pub/pcre/Public-Key
 
 # Refused by upstream, bug #675477
@@ -38,7 +42,7 @@ BuildArch:	noarch
 BuildRequires:  make
 BuildRequires:  git
 BuildRequires:	redhat-rpm-config
-BuildRequires:  gnupg2
+BuildRequires:  openpgpverify
 BuildRequires:  autoconf
 BuildRequires:  automake
 BuildRequires:  libtool
@@ -113,7 +117,7 @@ Static version of the mingw64-pcre library.
 
 
 %prep
-%{gpgverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
+%{openpgpverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
 %autosetup -S git_am -n pcre-%{version}
 autoreconf -vif
 
@@ -207,6 +211,9 @@ find $RPM_BUILD_ROOT -name "*.la" -delete
 
 
 %changelog
+* Fri Oct 23 2026 Jakub Jelen <jjelen@redhat.com> - 8.45-12
+- Verify upstream source signatures with openpgpverify
+
 * Thu Jul 16 2026 Fedora Release Engineering <releng@fedoraproject.org> - 8.45-11
 - Rebuilt for https://fedoraproject.org/wiki/Fedora_45_Mass_Rebuild
 
-- 
2.56.0

